### PR TITLE
Endtoend demo

### PR DESCRIPTION
**Pull request by gonzalozawa**
@mcgoughn could you please review the end to end demo?
Thanks,

### CODE REVIEW

File: www/casestudies/items/endtoend/assessment.php [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+include_once '../../../config.php';
+include_once 'includes/header.php';
+
+use LearnositySdk\Request\Init;
+use LearnositySdk\Utils\Uuid;
+
+
+$item_ref = Uuid::generate();
+$activity_id = 'NBS_Demo';
```
Comment (mcgoughn) [inline www/casestudies/items/endtoend/assessment.php]: Can you change this from NBS demo to something generic?

File: www/casestudies/items/endtoend/assessment.php [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    $session_state = 'resume';
+} else {
+    $session_id = Uuid::generate();
+    $session_state = 'initial';
+}
+
+
+$request = array(
+    'activity_id'    => $activity_id,
+    'name'           => 'Items API demo - assess activity',
```
Comment (mcgoughn) [inline www/casestudies/items/endtoend/assessment.php]: and this to something a bit more descriptive?

File: www/casestudies/items/endtoend/assessment.php [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+$Init = new Init('items', $security, $consumer_secret, $request);
+$signedRequest = $Init->generate();
+
+?>
+
+    <div class="jumbotron section">
+        <div class="overview">
+            <h1>End to End Demo – Assessment</h1>
+            <p>Here is a sample student assessment including the question created by the author.</p>
```
Comment (mcgoughn) [inline www/casestudies/items/endtoend/assessment.php]: +            <p>Here is a sample student assessment containing the questions created by the author in step 1.</p>

File: www/casestudies/items/endtoend/assessment.php [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    'state'          => 'initial',
+    'type'           => 'submit_practice',
+    'course_id'      => $courseid,
+    'session_id'     => $session_id,
+    'user_id'        => $studentid,
+    'items'          => getItemsArray(),
+    'assess_inline'  => true,
+    'config'         => [
+        'title' => 'Demo Activity',
+        'captureOnResumeError' => true,
```
Comment (mcgoughn) [inline www/casestudies/items/endtoend/assessment.php]: can actually probably remove this line.

File: www/casestudies/items/endtoend/authoring.php [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+<div class="section">        
+    <div class="row">
+        <div id="notifications" class="col-md-2"></div>
+        <div class="col-md-10">
+            <div id="learnosity-author"></div>
+        </div>
+    </div>  
+    <p class="text-right">
+        <br>
+        <a class="btn btn-primary btn-md" href="javascript:addNextQuestion();">Add more</a>
```
Comment (mcgoughn) [inline www/casestudies/items/endtoend/authoring.php]: I'd remove the javascript from the href link, and instead add it via an on('click') handler in JS.

File: www/casestudies/items/endtoend/authoring.php [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+        return Math.floor((1 + Math.random()) * 0x10000)
+          .toString(16)
+          .substring(1);
+      }
+      return s4() + s4() + '-' + s4() + '-' + s4() + '-' +
+        s4() + '-' + s4() + s4() + s4();
+    }
+
+    function saveItemID(itemID) {
+        var idx = itemIDs.indexOf(itemID);
```
Comment (mcgoughn) [inline www/casestudies/items/endtoend/authoring.php]: you can use Jquery and use .inArray() instead to simplify this check, and keep it in the "if" statement

File: www/casestudies/items/endtoend/authoring.php [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+        }
+    }
+    
+    function removeItem(itemID) {
+        itemIDs.splice(itemIDs.indexOf(itemID), 1);
+        $("#ItemNotification" + itemID).remove();
+    }
+
+    function editItem(itemID) {        
+        initOptions.request.reference = itemID;
```
Comment (mcgoughn) [inline www/casestudies/items/endtoend/authoring.php]: is this line necessary?

File: www/casestudies/items/endtoend/index.php [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+<div class="jumbotron section">
+    <div class="pull-right toolbar">
+        <ul class="list-inline">
+            <li data-toggle="tooltip" data-original-title="Toggle product overview box"><a href="#"><span class="glyphicon glyphicon-chevron-up jumbotron-toggle"></span></a></li>
+        </ul>
+    </div>
+    <h1>End to End Demo</h1>
+    <div class="section-intro">
+        <p>This is an End to End demo covering the main areas of Learnosity APIs.<p>
+        <p>Starting <strong>Authoring</strong> questions to turn them into an <strong>Assessment</strong> and finally <strong>Reporting</strong> to review the results and provide feedback.</p>
```
Comment (mcgoughn) [inline www/casestudies/items/endtoend/index.php]: <p>Starting with <strong>Authoring</strong> questions to turn them into an <strong>Assessment</strong> and finally <strong>Reporting</strong> to review the results and provide feedback.</p>

File: www/casestudies/items/endtoend/feedback.php [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+
+$Init = new Init('reports', $security, $consumer_secret, $request);
+$signedRequest = $Init->generate();
+
+?>
+
+    <div class="jumbotron section">       
+        <div class="overview">
+            <h1>End to End Demo – Reporting Feedback</h1>
+            <p>Using Reports API on the left side you can see the student answers.</p>
```
Comment (mcgoughn) [inline www/casestudies/items/endtoend/feedback.php]: +            <p>Using Reports API on the left side you can see the student answers.</p>
+            <p>On the right side of the page the teacher can provide additional subjective feedback to the student, alongside the existing objective scoring.</p>

**Comment by gonzalozawa** [comment]
@mcgoughn I have done the changes. Please let me know your comments.

**Pull request merged**